frontend/src/machines/billingMachine.ts

```diff
@@ -1,3 +1,137 @@
-import { createMachine } from "xstate";
+import { createMachine, assign } from "xstate";
 
-export const billingMachine = createMachine({});
+type Billing = {
+  id: string;
+  title: string;
+  amount: number;
+};
+
+interface Context {
+  billings: Billing[];
+  isEditingForm: boolean;
+  isCreatingForm: boolean;
+}
+
+const billingMachine = createMachine<Context>({
+  id: "billing",
+  initial: "idle",
+  context: {
+    billings: [],
+    isCreatingForm: false,
+    isEditingForm: false,
+  },
+  states: {
+    idle: {
+      on: {
+        FETCHINGBILLINGS: {
+          target: "gettingBilling",
+        },
+      },
+    },
+    gettingBilling: {
+      on: {
+        FETCHINGBILLINGSSUCCES: {
+          target: "gettingBillingOK",
+        },
+        FETCHINGBILLINGSERROR: {
+          target: "gettingBillingError",
+        },
+      },
+    },
+    gettingBillingOK: {
+      on: {
+        ACTIVATEBILLINGFORM: {
+          target: "billingFormIdle",
+        },
+      },
+    },
+    gettingBillingError: {
+      on: {
+        REFETCHINGBILLINGS: {
+          target: "gettingBilling",
+        },
+      },
+    },
+    billingFormIdle: {
+      always: [
+        {
+          cond: "creatingForm",
+          target: "#firstStep",
```

Comment:
Mau tanya, ini targetnya kenapa berdasarkan ID ya ? kenapa gak berdasarkan nama statenya ? ada benefitnya kah ?

Reply:
Good question, jadi kenapa aku pake hash `#`, karena kalau kita mau navigate ke state yang beda level itu ga bisa langsung dengan tulis nama states nya, in this case kita gabisa langsung `target: "firstStep"`.  

Nah, dari si XSTATE udah provide dengan cara target si state yang beda level nya dengan pake `#` plus nama state nya. 

Biar lebih kebayang struktur state nya kan gini ya.

````
Level 1 => BillingFormIdle

Level 1 => BillingFormReady
      Level 2 => FirstStep
````

Additional Notes*
Jangan lupa buat kasih id dari si State nya, otherwise `#` nya bakal useless

Reply:
Kalau akses parent statenya dulu bisa gak ya ? misalnya `billingFormReady.firstStep` ? atau sebenernya bisa langsung ke `billingFormReady` aja, soalnya initialnya kan ke `firstStep`

Reply:
I see, in this case aku rasa OK.